lib/dao.js

```diff
@@ -468,6 +468,18 @@ DataAccessObject.updateOrCreate = DataAccessObject.upsert = function upsert(data
   var Model = this;
   var connector = Model.getConnector();
 
+  if (Array.isArray(data)) {
+    var isBulkCreate = data.every(function(item){
```

Comment:
I really think we should add comments to document things like this. Like there seems to be some sort of backwards compat thing we're coding for but it's not explained. I expect something like:

```
// if we get an array of instances without id, call create - backwards compat
```

or something along those lines in the code.